dali/operators/random/rng_base.h

```diff
@@ -53,18 +55,37 @@ class RNGBase : public Operator<Backend> {
       rng_ = rng;
     } else {
       static_assert(std::is_same_v<Backend, GPUBackend>);
-      DALI_FAIL("Checkpointing is not implemented for GPU random operators. ");
+      const auto &states_cpu = cpt.CheckpointState<std::shared_ptr<curandState>>();
```

Comment:
Isn't it `states_gpu` :P? 
I see that the Serialize and DeserializeCheckpoint do the copies between device and host.

Reply:
Ooops! The design has changed and the old name remained :P Thanks!